### ISSUE TITLE
@apply creates rules with +1 class specificity (shady dom)

### ISSUE THREAD

**Issue by zhaoz** [Polymer/polymer#2439]
If I use a @apply, the CSS  gets moved out into a separate css block with a more specific rule. This is in polymer 1.0

```CSS
:root {
  --mixin: {
    background-color: blue;
  }
}

:host {
  @apply(--mixin);
}

:host.red {
  background-color: red;
}
```

The element being styled will be blue background, even if it has the class 'red' on it.

This is non-obvious reading the code, and not the easiest to fix since you have to add junk selectors to everything to make it be more specific.

From what I can tell the problem is two fold.
1. @apply creates a separate css block with a selector that is more specific.
2. The extra css block is appended at the end, which means although it has the same specificity, the CSS cascade overrides the :host.red rules.

**Comment by madshensel** [comment]
+1 on getting this fixed. I'm currently using the :host:not(.red) {...} as a workaround.

**Comment by noahlaux** [comment]
+1 I'm having this issue as well

**Comment by zhaoz** [comment]
@sorvell Can you take a look?

**Comment by ebidel** [comment]
@zhaoz just as a note, it should be `:host(.red)` and not `:host.red`. https://www.polymer-project.org/0.5/articles/styling-elements.html#element-defined-styles

**Comment by MeTaNoV** [comment]
oh god, I think I have the same derivated issue...

**Comment by MeTaNoV** [comment]
Can a CSS expert confirm: http://metanov.github.io/gold-password-input/ (on the demo page)
I have an addon that compute and show the strength score of the password, it get however in 3 cases overridden by a selector defined in `paper-input-container` (when focused and therefore addons are by definition highlighted)
@azakus any idea? :smile:

**Comment by arthurevans** [comment]
This issue appears to also occur with `var`:

http://jsbin.com/qetoho/1/edit?html,console,output

The `.container.loading` should win here, but instead we see the background color from the (less-specific)  first rule getting applied.

Changing the `var(--foo-background-color)` to `red` makes the jsbin work properly.

**Comment by kevinpschaaf** [comment]
May be fixed per #3413... need to re-test.

**Comment by arthurevans** [comment]
Running my jsbin above against master, it now appears to work correctly:

http://jsbin.com/qetoho/5/edit?html,console,output

Likewise, an @apply use case similar to the original report now works as expected:

http://jsbin.com/fusomu/edit?html,console,output

**Comment by zhaoz** [comment]
Checking in here, can this bug be closed? its been 25 days since it has appeared to work correctly.

**Comment by azakus** [comment]
Fixed with #3413
